pages/api_integration-full-node-streaming.md

```diff
@@ -332,6 +420,94 @@ func (c *GrpcClient) ProcessMatchPerpetualLiquidation(
 
 </details>
 
+### StreamSubaccountUpdate
+This message is used to update subaccount balances and positions.
+
+The initial message for a subaccount will have `snapshot` set to `true`. This message contains the full state of the subaccount. All updates should be ignored until the snapshot is received. Subsequent updates will contain updates to the positions and balances of the subaccount. They should be merged in with the existing state of the subaccount.
+
+<details>
+
+<summary>Code Snippet</summary>
+
+```go
+type SubaccountId struct {
+    Owner  string
+    Number int
+}
+
+type SubaccountPerpetualPosition struct {
+    PerpetualId int
+    Quantums    int
+}
+
+type SubaccountAssetPosition struct {
+    AssetId  int
+    Quantums int
+}
+
+type SubaccountState struct {
+    SubaccountId       SubaccountId
+    PerpetualPositions map[int]SubaccountPerpetualPosition
+    AssetPositions     map[int]SubaccountAssetPosition
+}
+
+func (c *GrpcClient) ProcessSubaccountUpdate(
+    subaccountUpdate *satypes.StreamSubaccountUpdate,
+    subaccountMap map[satypes.SubaccountId]SubaccountState,
+) {
+    // Extract the subaccount ID from the update
+    subaccountId := *subaccountUpdate.SubaccountId
+
+    // Check if this is a snapshot
+    if subaccountUpdate.Snapshot {
+        // Replace the entire subaccount state with the snapshot data
+        subaccountState := SubaccountState{
+            SubaccountId:       subaccountId,
+            PerpetualPositions: make(map[int]SubaccountPerpetualPosition),
+            AssetPositions:     make(map[int]SubaccountAssetPosition),
+        }
+
+        // Populate perpetual positions from snapshot
+        for _, perpPositionUpdate := range subaccountUpdate.UpdatedPerpetualPositions {
+            subaccountState.PerpetualPositions[perpPositionUpdate.PerpetualId] = *perpPositionUpdate
+        }
+
+        // Populate asset positions from snapshot
+        for _, assetPositionUpdate := range subaccountUpdate.UpdatedAssetPositions {
+            subaccountState.AssetPositions[assetPositionUpdate.AssetId] = *assetPositionUpdate
+        }
+
+        // Update the map with the new snapshot state
+        subaccountMap[subaccountId] = subaccountState
+    } else {
+        // If not a snapshot, retrieve or initialize the current subaccount state
+        subaccountState, exists := subaccountMap[subaccountId]
+        if !exists {
+            subaccountState = SubaccountState{
+                SubaccountId:       subaccountId,
+                PerpetualPositions: make(map[int]SubaccountPerpetualPosition),
+                AssetPositions:     make(map[int]SubaccountAssetPosition),
+            }
+        }
+
+        // Update perpetual positions
+        for _, perpPositionUpdate := range subaccountUpdate.UpdatedPerpetualPositions {
```

Comment:
For this implementation, we never delete entries in the map if the values reach zero right? so the size of the map is strictly increasing. should we delete entries in the map if the values are zeroed out?

Reply:
done